src/GraphQL/Resolver/ItemResolver.php

```diff
@@ -0,0 +1,237 @@
+     *
+     * @param \Overblog\GraphQLBundle\Definition\Argument|array $args
+     * @param string|null $contentTypeIdentifier
+     *
+     * @return Item
+     *
+     * @throws \GraphQL\Error\UserError if $contentTypeIdentifier was specified, and the loaded item's type didn't match it
+     * @throws \GraphQL\Error\UserError if no argument was provided
+     */
+    public function resolveItemOfType($args, $contentTypeIdentifier): Item
+    {
+        if (isset($args['id'])) {
+            $item = $this->itemFactory->fromContent(
+                $this->contentLoader->findSingle(new Query\Criterion\ContentId($args['id']))
+            );
+        } elseif (isset($args['contentId'])) {
+            $item = $this->itemFactory->fromContent(
+                $content = $this->contentLoader->findSingle(new Query\Criterion\ContentId($args['contentId']))
+            );
+        } elseif (isset($args['remoteId'])) {
+            $item = $this->itemFactory->fromContent(
+                $this->contentLoader->findSingle(new Query\Criterion\RemoteId($args['remoteId']))
+            );
+        } elseif (isset($args['locationId'])) {
+            $item = $this->itemFactory->fromLocation(
+                $this->locationLoader->findById($args['locationId'])
+            );
+        } elseif (isset($args['locationRemoteId'])) {
+            $item = $this->itemFactory->fromLocation(
+                $this->locationLoader->findByRemoteId($args['locationRemoteId'])
+            );
+        } else {
+            throw new UserError('Missing required argument id, remoteId or locationId');
+        }
+
+        $contentType = $item->getContentInfo()->getContentType();
+
+        if ($contentType->identifier !== $contentTypeIdentifier) {
+            throw new UserError("Content {$item->getContentInfo()->id} is not of type '$contentTypeIdentifier'");
+        }
+
+        return $item;
+    }
+
+    public function resolveItemFieldValue(Item $item, $fieldDefinitionIdentifier)
+    {
+        return Field::fromField($item->getContent()->getField($fieldDefinitionIdentifier));
+    }
+
+    public function resolveItemRelationFieldValue(Field $field, $multiple = false)
+    {
+        $destinationContentIds = $this->getContentIds($field);
+
+        if (empty($destinationContentIds) || array_key_exists(0, $destinationContentIds) && null === $destinationContentIds[0]) {
+            return $multiple ? [] : null;
+        }
+
+        // @todo use Collection Filter Builder
+        $contentItems = $this->contentLoader->find(new Query(
+            ['filter' => new Query\Criterion\ContentId($destinationContentIds)]
+        ));
+
+        if ($multiple) {
+            return array_map(
+                function ($contentId) use ($contentItems) {
+                    return $contentItems[array_search($contentId, array_column($contentItems, 'id'))];
+                },
+                $destinationContentIds
+            );
```

Comment:
At the moment whole expression could be reduced to `return $contentItems` Did you mean 

```
return array_map(
    function (Content $content): Item {
        return $this->itemFactory->fromContent($content);
    },
    $contentItems
);     
```

?

Reply:
If I recall correctly (I wrote that a while ago) the goal was to return the items in the order they are defined in the relation field. And if I recall correctly (again), the search result doesn't respect the order the id are passed in.

Reply:
> the search result doesn't respect the order the id are passed in.

True, especially in multi node setup.